libraries/AP_ESC_Telem/AP_ESC_Telem_Backend.cpp

```diff
@@ -0,0 +1,59 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "AP_ESC_Telem_Backend.h"
+#include <AP_HAL/AP_HAL.h>
+#include <AP_Math/AP_Math.h>
+
+// return the average motor frequency in Hz for dynamic filtering
+float AP_ESC_Telem_Backend::get_average_motor_frequency_hz() const
+{
+    float motor_freq = 0.0f;
+    const uint32_t now = AP_HAL::millis();
+    uint8_t valid_escs = 0;
+
+    // average the rpm of each motor and convert to Hz
+    for (uint8_t i = 0; i < get_num_motors(); i++) {
+        uint32_t last_telem = have_telem_data(i);
+        uint16_t rpm;
+        if (last_telem > 0 && (now - last_telem < 1000) && get_rpm(i, rpm)) {
+            valid_escs++;
+            motor_freq = rpm / 60.0f; // for better results we should slew here
```

Comment:
How about replacing this with a multiplication?

Reply:
I really don't want to here - its a recurring decimal and 60 is obviously converting from rpm to hz